lib/addons/knative-operator/index.ts

```diff
@@ -0,0 +1,69 @@
+import { Construct } from 'constructs';
+import { ClusterAddOn, ClusterInfo } from "../../spi";
+import {dependable, loadExternalYaml} from "../../utils";
+import { KubectlProvider } from "../helm-addon/kubectl-provider";
+
+/**
+ * 
+ */
+export interface KnativeOperatorProps {
+    /**
+     * The namespace to install Knative in
+     * @default default
+     */
+    namespace?: string;
+
+    /**
+     * The name to be assigned to given to the Knative operator
+     * @default knative-operator
+     */
+    name?: string;
+
+    /**
+     * The version of the KNative Operator to use
+     * @default v1.8.1
+     */
+    version?: string;
+
+    /**
+     * 
+     * @default 'https://github.com/knative/operator/releases/download/knative'
+     */
+    base_url?: string;
+}
+
+const defaultProps = {
+    name: 'knative-operator',
+    namespace: 'default',
```

Comment:
How common it is to place the operator into the default namespace vs a custom namespace like knative?
If a customer defines the namespace to be custom, how do you see they will approach namespace creation, provided the add-on has no logic to do it?

Reply:
Based on my deduction and brief experimentation, the Knative Operator needs to be installed to the default namespace so the associated CRDs can be used in any other namespace.

I know with the YAML deployment, you have to deploy the CRDs per namespace and the Operator avoids that repetition.